### PR TITLE
Do not clamp grouping columns

### PR DESCRIPTION
**Pull request by vserret**
see issue #359

### CODE REVIEW

File: sdk/opendp/smartnoise/sql/private_rewriter.py [hunk 1]
```diff
@@ -247,45 +247,54 @@ def per_key_clamped(self, query):
...
             cols += [exp]
         for col in cols:
             colname = col.name
-            minval = None
-            maxval = None
-            sym = col.symbol(relations)
-            if do_clamp and sym.valtype in ["float", "int"] and not sym.unbounded:
-                minval = sym.minval
-                maxval = sym.maxval
-                if minval is None or sym.is_key:
```
Comment (vserret) [inline sdk/opendp/smartnoise/sql/private_rewriter.py:274]: I do not understand this line: why `maxval is None` does not appear here ?

File: sdk/opendp/smartnoise/sql/private_rewriter.py [hunk 2]
```diff
@@ -247,45 +247,54 @@ def per_key_clamped(self, query):
...
+            if not do_clamp:
+                return None, None
+            if query.agg:
+                grouping_colnames = [col.name for col in query.agg.groupedColumns()]
+                if colname in grouping_colnames:
+                    return None, None
+            minval = None
+            maxval = None
+            sym = col.symbol(relations)
+            if sym.valtype in ["float", "int"] and not sym.unbounded:
```
Comment (vserret) [inline sdk/opendp/smartnoise/sql/private_rewriter.py:272]: I am not sure of this line
See my previous comment

File: sdk/opendp/smartnoise/sql/private_rewriter.py (hunk 1 above)
Comment (joshua-oss) [reply sdk/opendp/smartnoise/sql/private_rewriter.py:274]: That's an oversight, it should be checking maxval.  I just pushed the change

File: sdk/opendp/smartnoise/sql/private_rewriter.py (hunk 2 above)
Comment (joshua-oss) [reply sdk/opendp/smartnoise/sql/private_rewriter.py:272]: This change looks good.  We should add a unit test to make sure it's behaving as intended, and to protect against regressions in the future.

**Review by joshua-oss — APPROVED** [review]
Added unit test and approved, thanks for the fix!!

**Pull request merged**